infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/logreader/RoutingQueuesSnapshotReader.java

```diff
@@ -269,24 +269,24 @@ private void waitForData() {
     // Check if the Opaque Entry contains an End Marker, denoting the end of snapshot sync data
     private boolean isEndMarker(OpaqueEntry opaqueEntry) {
         CorfuStoreEntry<Queue.RoutingQSnapSyncHeaderKeyMsg, Queue.RoutingQSnapSyncHeaderMsg, Message> entry =
-                getFullSyncTxnHeader(opaqueEntry);
+                getSnapSyncTxnHeader(opaqueEntry);
         if (entry == null) {
             return false;
         }
 
-        log.info("isEndMarker is called from next of snapshot reader. full sync request {} {}", entry.getKey(), entry.getPayload());
+        log.info("isEndMarker is called from next of snapshot reader. snapshot sync request {} {}", entry.getKey(), entry.getPayload());
         // The protocol is the end marker is represented by the Full Sync Request Id but negated.
-        return (entry.getPayload().getSnapshotStartTimestamp() == -requestFullSyncId);
+        return (entry.getPayload().getSnapshotStartTimestamp() == -requestSnapSyncId);
```

Comment:
Is this - intentional?

Reply:
yes - we are using the negative value as a way to both identify the request and also mark the completion. Doing it this way avoids having an extra payload field.

Reply:
it is safe since java does not have unsigned long